lib/monitor.js

```diff
@@ -239,7 +241,22 @@ internals.Monitor.prototype._opsHandler = function (results) {
 
 internals.Monitor.prototype._wreckHandler = function (error, request, response, start, uri) {
 
-    this._dataStream.push(Utils.GreatWreck(error, request, response, start, uri));
+    var self = this;
```

Comment:
I don't really like this change. If you notice there isn't any `JSON.parse` anywhere else in the code as that is left to the individual reporters to deal with. Also the `try, catch` operation in this function will prevent this code from being optimized by v8.